pages/index.tsx

```diff
@@ -31,6 +32,7 @@ const useStyles = createStyles((theme) => ({
  * Homepage component.
  */
 export default function Homepage() {
+  const [blurNSFW, setBlurNSFW] = useState<boolean>(true)
```

Comment:
@indiehjaerta The functionality of this PR is working perfectly. Nicely done.

You even mentioned using a state library in your PR. I'd like to avoid another lib, and just continue to use React Context. 

I should have been more clear in the ticket about this, but I would like you to move the `blurNSFR` state out of `Homepage()` and move it to context provider in `<RedditProvider />`. 

Otherwise, this PR is good to go. Thanks!